rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/ArtifactDownloaderImpl.java

```diff
@@ -24,20 +24,27 @@ public ArtifactDownloaderImpl(ClouddriverService clouddriverService) {
     this.clouddriverService = clouddriverService;
   }
 
-  public void downloadArtifact(Artifact artifact, Path targetFile) throws IOException {
-    try (OutputStream outputStream = Files.newOutputStream(targetFile)) {
-      Response response =
-          retrySupport.retry(() -> clouddriverService.fetchArtifact(artifact), 5, 1000, true);
-      if (response.getBody() != null) {
-        try (InputStream inputStream = response.getBody().in()) {
-          IOUtils.copy(inputStream, outputStream);
-        } catch (IOException e) {
-          throw new IOException(
-              String.format(
-                  "Failed to read input stream of downloaded artifact: %s. Error: %s",
-                  artifact, e.getMessage()));
-        }
+  public InputStream downloadArtifact(Artifact artifact) throws IOException {
+    Response response =
+        retrySupport.retry(() -> clouddriverService.fetchArtifact(artifact), 5, 1000, true);
+    if (response.getBody() == null) {
+      throw new IOException("Failure to fetch artifact: empty response");
+    } else {
+      try (InputStream inputStream = response.getBody().in()) {
```

Comment:
I don't think we'll want a `try-with-resources` in this function, as it's directly returning the `InputStream` which should remain open for the caller (whereas `try-with-resources` will close it).

I think what this function should do is just `return response.getBody().in()` after the null check.  Then callers (including the below `downloadArtifactToFile`) should call this function as:
```
try (InputStream inputStream = downloadArtifact(artifact)) {
// ...use inputStream
}
```
That means we'll have nested `try-with-resources` blocks in `downloadArtifactToFile` but that's fine and the usual way of solving this.

Reply:
Ah, that makes sense--updated. I have the catches nested as well, to preserve the logic of the error messages Maggie recently added.